Sources/AardvarkReveal/RevealAttachmentGenerator.swift

```diff
@@ -0,0 +1,518 @@
+///                                 │
+///    .────────────────────────────┴────────────────────────────.
+///   (    revealAttachmentGeneratorDidFinishCapturingAppState    )
+///    `────────────────────────────┬────────────────────────────'
+///                                 │                                      .───────────────────────────.
+///                                 ├──────────────────────────────────▶  (       completion(nil)       )
+///                                 │            Failed to                 `───────────────────────────'
+///                        Captured │          Capture State
+///                       App State │
+///                                 ▼
+///   ┌───────────────────────────────────────────────────────────┐
+///   │                                                           │
+///   │                    Capturing Snapshots                    │
+///   │                                                           │
+///   │  .─────────────────────────────────────────────────────.  │
+///   │ ( revealAttachmentGeneratorDidCaptureMainScreenSnapshot ) │
+///   │  `─────────────────────────────────────────────────────'  │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///                                 ▼
+///   ┌───────────────────────────────────────────────────────────┐
+///   │                                                           │
+///   │                   Bundling Reveal File                    │
+///   │                                                           │
+///   └───────────────────────────────────────────────────────────┘
+///                                 │
+///    .────────────────────────────┴────────────────────────────.
+///   (   revealAttachmentGeneratorDidFinishBundlingRevealFile    )
+///    `────────────────────────────┬────────────────────────────'
+///                                 ▼
+///                   .───────────────────────────.
+///                  (         completion          )
+///                   `───────────────────────────'
+/// ```
+///
+/// It is important to avoid changing anything while the app state is being captured (during the `Capturing App State`
+/// phase). Once the app state is captured, the generator will begin capturing snapshots of each drawable object (views,
+/// layers, etc.) on the screen. Once the generator has captured a snapshot of the main screen (marked by a call to the
+/// delegate's `revealAttachmentGeneratorDidCaptureMainScreenSnapshot` method), it is okay to display a waiting
+/// indicator **as long as all of the existing views/layers remain in the hierarchy** (for example, you can display a
+/// new window on top of the main window or present a modal view controller, depending on the use case). Once the
+/// generator has completed bundling the Reveal file, user interaction can be restored.
```

Comment:
one hell of a comment. well done.

Reply:
Thanks, this took quite a while to write up 😂